src/StorageSync/StorageSync/RegisteredServer/RegisterServerCommand.cs

```diff
@@ -0,0 +1,195 @@
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+           Position = 1,
+           ParameterSetName =StorageSyncParameterSets.StringParameterSet,
+           Mandatory = true,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = HelpMessages.StorageSyncServiceNameParameter)]
+        [StorageSyncServiceCompleter]
+        [ValidateNotNullOrEmpty]
+        [Alias(StorageSyncAliases.ParentNameAlias)]
+        public string StorageSyncServiceName { get; set; }
+
+        [Parameter(
+           Position = 0,
+           ParameterSetName = StorageSyncParameterSets.ObjectParameterSet,
+           Mandatory = true,
+           ValueFromPipeline = true,
+           HelpMessage = HelpMessages.StorageSyncServiceObjectParameter)]
+        [ValidateNotNullOrEmpty]
+        [Alias(StorageSyncAliases.StorageSyncServiceAlias)]
+        public PSStorageSyncService ParentObject { get; set; }
+
+        [Parameter(
+          Position = 0,
+          ParameterSetName = StorageSyncParameterSets.ParentStringParameterSet,
+          Mandatory = true,
+          ValueFromPipelineByPropertyName = true,
+          HelpMessage = HelpMessages.StorageSyncServiceParentResourceIdParameter)]
+        [ValidateNotNullOrEmpty]
+        [ResourceIdCompleter(StorageSyncConstants.StorageSyncServiceType)]
+        [Alias(StorageSyncAliases.StorageSyncServiceIdAlias)]
+        public string ParentResourceId { get; set; }
+
+
+        [Parameter(Mandatory = false, HelpMessage = HelpMessages.AsJobParameter)]
+        public SwitchParameter AsJob { get; set; }
+
+        protected override string Target => StorageSyncServiceName ?? ParentObject?.StorageSyncServiceName ?? ParentResourceId;
+
+        protected override string ActionMessage => $"Create a new Registered Server for Storage Sync Service {StorageSyncServiceName ?? ParentObject?.StorageSyncServiceName ?? ParentResourceId}";
+
+        public override void ExecuteCmdlet()
+        {
+            if (ShouldProcess(Target, ActionMessage))
+            {
+                base.ExecuteCmdlet();
+
+                ExecuteClientAction(() =>
+                {
+                    var parentResourceIdentifier = default(ResourceIdentifier);
+
+                    if (!string.IsNullOrEmpty(ParentResourceId))
+                    {
+                        parentResourceIdentifier = new ResourceIdentifier(ParentResourceId);
+
+                        if (!string.Equals(StorageSyncConstants.StorageSyncServiceType, parentResourceIdentifier.ResourceType, System.StringComparison.OrdinalIgnoreCase))
+                        {
+                            throw new PSArgumentException($"Invalid Argument {nameof(ParentResourceId)}", nameof(ParentResourceId));
+                        }
+                    }
+
+                    var resourceGroupName = ResourceGroupName ?? ParentObject?.ResourceGroupName ?? parentResourceIdentifier?.ResourceGroupName;
```

Comment:
see above - apply to all checks here - only the Subscription check appears to be a necessary check